### ISSUE TITLE
Makefile checker reports annoying message when editing a Makefile snippet which defines no targets

### ISSUE THREAD

**Issue by ackalker** [flycheck/flycheck#572]
When editing a Makefile snippet (a file for use with `include Makefile.foo` in another Makefile) marked with the property line `# -*- mode: makefile-gmake; -*-` for syntax highlighting (and Flycheck, of course :-) ), I get distracted by Flycheck repeatedly displaying the message:

```
Suspicious state from syntax checker make: Checker make returned non-zero exit code 2, but no errors from output: make: *** No targets.  Stop.
```

It is not uncommon for Makefile snippets to define no targets, they are often used as configuration files or function libraries.
A workaround is to add a dummy target (with an empty recipe) to shut up the messages, but this isn't always possible: depending on where in the original Makefile the snippet is included, this dummy target could become the default target (made when running just `make`), preventing any further processing.
Adding the 'real' default target as a prerequisite of the dummy target may be confusing to others who are not using Flycheck, and it presents extra maintenance work (and another way for insidious bugs to creep in) when the default target changes.

Please do not report messages like the above when a Makefile (snippet) defines no targets.

**Comment by lunaryorn** [comment]
@ackalker How can I tell `make` not to error for files without targets?

**Comment by ackalker** [comment]
Getting rid of the error appears to be quite difficult. I've tried both `make -s` (silent) and `make -q` ('question` mode). Neither prevents this particular error message.

A hacky workaround could be to generate a temporary Makefile, say `tmpmakefile` (any unique name) with the contents:
```make
dummytarget: # yet another generated unique name

include Makefile
```
and have the checker use that temporary Makefile instead of the real one: `make -f tmpmakefile [other options`.

But maybe the only absolutely reliable way of dealing with this is to no longer treat this behaviour as suspicious (but only for the make checker of course) and to not show any messages.

**Comment by ackalker** [comment]
Tested, works well with GNU make: `make -q -f tmpmakefile`

- Missing Makefile => no error
- Empty Makefile => no error
- Makefile with syntax error => error, message has correct line number in Makefile.

**Comment by lunaryorn** [comment]
@ackalker I don't like that “solution”, and it's not easy to implement in current Flycheck anyway.

I'd rather disable the make checker if the file has no targets…

**Comment by ackalker** [comment]
It would be sad to lose syntax checking on Makefile snippets with no targets.
Given that syntax errors from `include`d snippets get filtered out when editing the main Makefile, they will only show up when actually running `make`.

It's too bad that `make` errors out when there are no targets. IMHO it should just conclude that any targets which need to be remade (i.e. none) are up to date. Sadly, reality (historical precedent I assume) is different.

**Comment by lunaryorn** [comment]
@ackalker Well, it's a trade off.  YMMV, but I do not want to add complex code just to work around broken behaviour of a single syntax checker.  And anyhow, #573 is just another reason to remove make support.

That is, unless there's a way to reliably switch make into a “lint-only” mode, of course.  I don't know whether there's a way, though, and I must admit, that I really don't feel like researching that.  I simply don't use `make` often enough to have much personal interest in this syntax checker.

If you'd take that task, I'd be happy, otherwise I'll just remove make this weekend.

**Comment by lunaryorn** [comment]
@ackalker I removed the make checker.  Closing.
